src/contributing.md

```diff
@@ -0,0 +1,140 @@
+# Contributing guidelines
+
+## Generic Reusable Component Design
+
+Define the component's purpose. Narrow down the scope as much as possible.
+
+Avoid assumptions about its parent/owner.
```

Comment:
Maybe we could provide an example:

SomeItem.qml
```qml
Item {
  Layout.fillWidth: true
  Layout.fillHeight: true
}
```
If `SomeItem` is not used in layout it would end up with warnings. I've seen it in some places in our code already.

Reply:
I provided examples in a separate file to keep this as a practical quick checklist and added redirection links, please have a look at the new structure.